extension/grpcindexer/external_subscriber_server.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package grpcindexer
+
+import (
+	"context"
+	"log"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/extension/indexer"
+	"github.com/ava-labs/hypersdk/pebble"
+
+	avametrics "github.com/ava-labs/avalanchego/api/metrics"
+	pb "github.com/ava-labs/hypersdk/proto"
+	hstorage "github.com/ava-labs/hypersdk/storage"
+)
+
+type ParserFactory func(networkID uint32, chainID ids.ID, genBytes []byte) (chain.Parser, error)
+
+type ExternalSubscriberServer struct {
+	pb.ExternalSubscriberServer
+	indexer       indexer.Indexer
+	parser        chain.Parser
+	parserFactory ParserFactory
+	Logger        *log.Logger
+}
+
+func NewExternalSubscriberServer(stdDBDir string, logger *log.Logger, parserFactory ParserFactory) *ExternalSubscriberServer {
+	stdDB, err := hstorage.New(pebble.NewDefaultConfig(), stdDBDir, "db", avametrics.NewLabelGatherer("standardIndexer"))
+	if err != nil {
+		logger.Fatalln("Failed to create DB for standard indexer")
+	}
+	return &ExternalSubscriberServer{
+		indexer:       indexer.NewDBIndexer(stdDB),
```

Comment:
Can we replace this with the splitter design that we discussed? Instead of adding the `indexer.NewDBIndexer` here, the external subscriber server should split the input to a slice of accepted subscribers that it knows nothing about.